bentoml/cli/deployment.py

```diff
@@ -213,3 +231,255 @@ def check_deployment_status(archive_path, platform, region, stage, api_name):
+    items = tag.split(':')
+
+    if len(items) > 2:
+        raise BentoMLException("More than one ':' appeared in tag '%s'" % tag)
+    elif len(items) == 1:
+        return tag, 'latest'
+    else:
+        if not items[0]:
+            raise BentoMLException("':' can't be the leading character")
+        if not items[1]:
+            raise BentoMLException("Please include value for the key %s" % items[0])
+        return items[0], items[1]
+
+
+def get_deployment_sub_command(cli):
+    @click.group()
+    def deploy():
+        pass
+
+    @deploy.command(
+        short_help="Apply deployment configuration",
+        context_settings=dict(ignore_unknown_options=True, allow_extra_args=True),
+    )
+    @click.argument('--bento-tag', type=click.STRING, required=True)
+    @click.argument("--deployment-name", type=click.STRING, required=True)
+    @click.option(
+        "--platform",
+        type=click.Choice(
+            ["aws_lambda", "gcp_function", "aws_sagemaker", "kubernetes", "custom"]
+        ),
+        required=True,
+        help="Target platform that Bento archive is going to deployed to",
+    )
+    @click.option("--namespace", type=click.STRING, help="Deployment's namespace")
+    @click.option("--labels", type=click.STRING, help="")
+    @click.option("--annotations", type=click.STRING)
+    @click.option(
+        '--region',
+        help="Name of the deployed region. For platforms: AWS_Lambda, AWS_SageMaker, GCP_Function",
+    )
+    @click.option(
+        '--stage', help="Stage is to identify. For platform:  AWS_Lambda, GCP_Function"
+    )
+    @click.option(
+        '--instance-type',
+        help="Type of instance will be used for inference. For platform: AWS_SageMaker",
+    )
+    @click.option(
+        '--instance-count',
+        help="Number of instance will be used. For platform: AWS_SageMaker",
+    )
+    @click.option(
+        '--api-name',
+        help="User defined API function will be used for inference. For platform: AWS_SageMaker",
+    )
+    @click.option(
+        '--kube-namespace',
+        help="Namespace for kubernetes deployment. For platform: Kubernetes",
+    )
+    @click.option('--replicas', help="Number of replicas. For platform: Kubernetes")
+    @click.option('--service-name', help="Name for service. For platform: Kubernetes")
+    @click.option('--service-type', help="Service Type. For platform: Kubernetes")
+    @click.option('--output', type=click.Choice(['json', 'yaml']), default='json')
+    def apply(
+        bento_tag,
+        deployment_name,
+        platform,
+        output,
+        namespace=None,
+        labels=None,
+        annotations=None,
+        region=None,
+        stage=None,
+        instance_type=None,
+        instance_count=None,
+        api_name=None,
+        kube_namespace=None,
+        replicas=None,
+        service_name=None,
+        service_type=None,
+    ):
+        bento_name, bento_verison = parse_bento_tag(bento_tag)
```

Comment:
let's use a validation callback for bento_tag argument https://click.palletsprojects.com/en/5.x/options/#callbacks-for-validation

that way here can be simply `bento_name, bento_verison = bento_tag.split(':')`